armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -310,3 +368,179 @@ def _insertDepletableNuclideKeys(c, blueprint):
+    """Component-like specification for a single layer in particle fuel"""
+
+    name = yamlize.Attribute(key="name", type=str)
+    material = yamlize.Attribute(type=str)
+    innerDiam = yamlize.Attribute(key="id", type=float, default=0)
+    od = yamlize.Attribute(key="od", type=float)
+    Tinput = yamlize.Attribute(type=float)
+    Thot = yamlize.Attribute(type=float)
+    # Need this to pick up flags like depletable
+    flags = yamlize.Attribute(type=str, default=None)
+
+    def construct(self, blueprint, matMods):
+        """Construct a sphere and assign to a parent"""
+        # Very similar to ComponentBlueprint.construct, maybe share?
+        runLog.debug(f"Constructing particle fuel layer {self.name}")
+        kwargs = self._conformKwargs(blueprint, matMods)
+        component = components.factory("sphere", [], kwargs)
+
+        # the component __init__ calls setType(), which gives us our initial guess at
+        # what the flags should be.
+        if self.flags is not None:
+            # override the flags from __init__ with the ones from the blueprint
+            component.p.flags = Flags.fromString(self.flags)
+        else:
+            # potentially add the DEPLETABLE flag. Don't do this if we set flags
+            # explicitly. WARNING: If you add flags explicitly, it will
+            # turn off depletion so be sure to add depletable to your list of flags
+            # if you expect depletion
+            if any(nuc in blueprint.activeNuclides for nuc in component.getNuclides()):
+                component.p.flags |= Flags.DEPLETABLE
+
+        if component.hasFlags(Flags.DEPLETABLE):
+            # depletable components, whether auto-derived or explicitly flagged
+            # need expanded nucs
+            _insertDepletableNuclideKeys(component, blueprint)
+
+        if any(nucDir.isHeavyMetal(nucName) for nucName in component.getNuclides()):
+            component.p.flags |= Flags.FUEL
+
+        return component
+
+    def _conformKwargs(self, blueprint, matMods) -> dict:
+        """Return dictionary of arguments to help with component construction"""
+        kwargs = {}
+        for attr in self.attributes:
+            key = attr.name
+            if key == "innerDiam":
+                key = "id"
+            elif key == "flags":
+                continue
+            val = attr.get_value(self)
+            if key == "material":
+                value = constructMaterial(val, blueprint, matMods, None)
+            else:
+                value = attr.get_value(self)
+            kwargs[key] = value
+        return kwargs
+
+
+class ParticleFuelSpec(yamlize.KeyedList):
+    """Specification for a single particle fuel type"""
+
+    item_type = _ParticleFuelLayer
+    key_attr = _ParticleFuelLayer.name
+    name = yamlize.Attribute(type=str)
+
+    def construct(self, blueprint, matMods):
+        """Produce a particle fuel instance that can be attached to a component"""
+        bounds = set()
+
```

Comment:
As mentioned above my dream here is that we can make a slightly more generic thing (`mixture`?) that would give you the capability needed while also potentially being applicable to highly related needs (e.g. the kinds of models mentioned in #503). 

We should discuss further.

Reply:
Excellent discussion post! I'll contribute some thoughts / musings